File: src/BudgetTracker.WebApi/TransferModels/Transaction.cs
namespace BudgetTracker.WebApi.TransferModels;

public class TransactionDto
{
    public string TransactionId { get; init; } = null!;
    public DateTime TransactionDate { get; init; }
    public decimal TransactionAmount { get; init; }
    public string? Description { get; set; }
    public string Currency { get; init; } = null!;
    public string TransactionTypeId { get; set; } = null!;
    public string TransactionTypeName { get; init; } = null!;
    public string CategoryId { get; set; } = null!;
    public string CategoryName { get; init; } = null!;
}

public class AddTransactionRequest
{
    public DateTime TransactionDate { get; init; }
    public decimal TransactionAmount { get; init; }
    public string Currency { get; init; } = null!;
    public string? Description { get; init; } = null!;
    public string TransactionTypeId { get; init; } = null!;
    public string CategoryId { get; init; } = null!;
}

File: client/src/api/models/transaction.ts
export interface TransactionTypeResponse {
  transactionTypeId: string;
  transactionTypeName: string;
  description: string;
  isDefaultType: boolean;
}

export interface AddTransactionTypeRequest {
  transactionTypeName: string;
  description: string;
}

export interface UpdateTransactionTypeRequest {
  transactionTypeId: string;
  transactionTypeName: string;
  description: string;
}

export interface DeleteTransactionTypeRequest {
  transactionTypeId: string;
}

export interface TransactionResponse {
  transactionId: string;
  transactionDate: Date;
  transactionAmount: number;
  description: string;
  currency: string;
  transactionTypeId: string;
  transactionTypeName: string;
  categoryId: string;
  categoryName: string;
}

export interface AddTransactionRequest {
  transactionDate: Date;
  transactionAmount: number;
  currency: string;
  description: string;
  transactionTypeId: string;
  categoryId: string;
}

export interface UpdateTransactionRequest {
  transactionId: string;
  transactionDate: Date;
  transactionAmount: number;
  currency: string;
  description: string;
  transactionTypeId: string;
  categoryId: string;
}

export interface DeleteTransactionRequest {
  transactionId: string;
}

export interface AddDefaultTransactionTypeRequest {
  transactionTypeName: string;
  description: string;
}

export interface DeleteDefaultTransactionTypeRequest {
  transactionTypeId: string;
}
